### ISSUE TITLE
NPE for artillery counterbattery fire against fled off board artillery

### ISSUE THREAD

**Issue by KaiserFalk** [MegaMek/megamek#4947]
### Environment
49.15 MekHQ
Windows 10
Java 17.0.8

### Description
Getting NPE during artillery phase due to enemy Thumpers targeting my Thumper which disengaged from off board artillery the turn before

### Files
[autosave.sav.gz](https://github.com/MegaMek/megamek/files/13629166/autosave.sav.gz)

[megamek.log](https://github.com/MegaMek/megamek/files/13629177/megamek.log)

Player name: Laker's Lancers

Customs:
[Custom Mechs.zip](https://github.com/MegaMek/megamek/files/13629180/Custom.Mechs.zip)

Reported previously when it happened with a Sniper turret:
https://github.com/MegaMek/megamek/issues/4790

**Comment by HammerGS** [comment]
Two different NPE's in here. 

```
15:54:15,439 ERROR [megamek.common.pathfinder.LongestPathFinder.LongestPathRelaxer] {Princess-precognition (Free Worlds League Integrated Allies)}
megamek.common.pathfinder.LongestPathFinder$LongestPathRelaxer.doRelax(LongestPathFinder.java:170) - Top Move Path uses more MPs than Move Path Candidate.
java.lang.IllegalStateException
	at megamek.common.pathfinder.LongestPathFinder$LongestPathRelaxer.doRelax(LongestPathFinder.java:170)
	at megamek.common.pathfinder.LongestPathFinder$LongestPathRelaxer.doRelax(LongestPathFinder.java:125)
	at megamek.common.pathfinder.AbstractPathFinder.run(AbstractPathFinder.java:266)
	at megamek.common.pathfinder.AbstractPathFinder.run(AbstractPathFinder.java:295)
	at megamek.client.bot.princess.PathEnumerator.recalculateMovesForWorker(PathEnumerator.java:242)
	at megamek.client.bot.princess.PathEnumerator.recalculateMovesFor(PathEnumerator.java:127)
	at megamek.client.bot.princess.Precognition.run(Precognition.java:364)
	at java.base/java.lang.Thread.run(Thread.java:833)
```
then

```
15:54:43,468 ERROR [megamek.MegaMek] {Packet Pump}
megamek.MegaMek.lambda$main$0(MegaMek.java:63) - Uncaught Exception Detected
java.lang.NullPointerException: Cannot invoke "megamek.common.Targetable.getPosition()" because "this.target" is null
	at megamek.common.weapons.ArtilleryWeaponIndirectFireHandler.handle(ArtilleryWeaponIndirectFireHandler.java:107)
	at megamek.server.GameManager.handleAttacks(GameManager.java:34407)
	at megamek.server.GameManager.handleAttacks(GameManager.java:34343)
	at megamek.server.GameManager.endCurrentPhase(GameManager.java:2484)
	at megamek.server.GameManager.changeToNextTurn(GameManager.java:2721)
	at megamek.server.GameManager.endCurrentTurn(GameManager.java:1571)
	at megamek.server.GameManager.receiveAttack(GameManager.java:13114)
	at megamek.server.GameManager.handlePacket(GameManager.java:763)
	at megamek.server.Server.handle(Server.java:1265)
	at megamek.server.Server$PacketPump.run(Server.java:111)
	at java.base/java.lang.Thread.run(Thread.java:833)
```

Related: 
https://github.com/MegaMek/megamek/issues/4790
https://github.com/MegaMek/megamek/issues/4794

**Comment by neoancient** [comment]
Duplicate of #4794, fixed by #4890. The fix will be part of the next release.
